projects/js-toolkit/packages/js-toolkit-core/src/project/liferayCli/Start.ts

```diff
@@ -5,11 +5,22 @@
 
 import LiferayJson from '../../schema/LiferayJson';
 import Project from './Project';
+import persist from './persist';
 
 export default class Start {
 	readonly port: number;
 
-	constructor(_project: Project, liferayJson: LiferayJson) {
+	constructor(project: Project, liferayJson: LiferayJson) {
+		this._project = project;
+
 		this.port = liferayJson.start?.port || 8081;
 	}
+
+	storePort(port: number): void {
+		(this as object)['port'] = port;
```

Comment:
Oh, yes, I solved that in a different way in another place of the code. Will change it.

This is the obvious solution to use when you want to update a `readonly` field, but it turns out that the TS guys have invented a [better way to do it](https://github.com/liferay/liferay-frontend-projects/pull/1004/files#diff-219c816c32294c23f66b1eda25b4bbec553ae2d18da169035ab7df2dc6a229f1R23)

Reply:
Oops, it's not that, it's that the options are polymorphic and the setter method allows passing in any option name.

We could try to type it, but it would be quite complex (I'm not even sure if it is possible because it would need generics based on a given parameter) and would make the code much less understandable for little gain.

Reply:
Whoa! I've removed the casting and tsc doesn't complain. Maybe because I added the switch after writing the casting and tsc is smart enough to infer that it is a valid dynamic access? :thinking: